### ISSUE TITLE
native client build fails with C99 warnings

### ISSUE THREAD

**Issue by gvoysey** [mozilla/DeepSpeech#441]
Following the instructions given [here](https://github.com/mozilla/DeepSpeech/tree/master/native_client), I was able to build tensorflow with CUDA 8 support on ubuntu 14.04.  Ultimate goal is to use the trained deepspeech model to do stt on wav files.

The command 
`bazel build -c opt //native_client:deepspeech`
fails: 

```
INFO: Found 1 target...
ERROR: /home/gvoysey/Projects/tensorflow/native_client/BUILD:3:1: C++ compilation of rule '//native_client:deepspeech' failed: gcc failed: error executing command 
  (cd /home/gvoysey/.cache/bazel/_bazel_gvoysey/d2979477669321f4c7bd28fc7c36db1c/execroot/tensorflow && \
  exec env - \
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections -MD -MF bazel-out/local-opt/bin/native_client/_objs/deepspeech/native_client/c_speech_features/c_speech_features.pic.d -fPIC -DEIGEN_MPL2_ONLY -DTENSORFLOW_USE_JEMALLOC -iquote . -iquote bazel-out/local-opt/genfiles -iquote external/bazel_tools -iquote bazel-out/local-opt/genfiles/external/bazel_tools -iquote external/eigen_archive -iquote bazel-out/local-opt/genfiles/external/eigen_archive -iquote external/local_config_sycl -iquote bazel-out/local-opt/genfiles/external/local_config_sycl -iquote external/jemalloc -iquote bazel-out/local-opt/genfiles/external/jemalloc -iquote external/protobuf -iquote bazel-out/local-opt/genfiles/external/protobuf -iquote external/gif_archive -iquote bazel-out/local-opt/genfiles/external/gif_archive -iquote external/jpeg -iquote bazel-out/local-opt/genfiles/external/jpeg -iquote external/com_googlesource_code_re2 -iquote bazel-out/local-opt/genfiles/external/com_googlesource_code_re2 -iquote external/farmhash_archive -iquote bazel-out/local-opt/genfiles/external/farmhash_archive -iquote external/highwayhash -iquote bazel-out/local-opt/genfiles/external/highwayhash -iquote external/png_archive -iquote bazel-out/local-opt/genfiles/external/png_archive -iquote external/zlib_archive -iquote bazel-out/local-opt/genfiles/external/zlib_archive -iquote external/local_config_cuda -iquote bazel-out/local-opt/genfiles/external/local_config_cuda -iquote external/grpc -iquote bazel-out/local-opt/genfiles/external/grpc -iquote external/nanopb_git -iquote bazel-out/local-opt/genfiles/external/nanopb_git -iquote external/gemmlowp -iquote bazel-out/local-opt/genfiles/external/gemmlowp -isystem native_client/c_speech_features -isystem bazel-out/local-opt/genfiles/native_client/c_speech_features -isystem native_client/kiss_fft130 -isystem bazel-out/local-opt/genfiles/native_client/kiss_fft130 -isystem external/bazel_tools/tools/cpp/gcc3 -isystem external/eigen_archive -isystem bazel-out/local-opt/genfiles/external/eigen_archive -isystem external/jemalloc/include -isystem bazel-out/local-opt/genfiles/external/jemalloc/include -isystem external/protobuf/src -isystem bazel-out/local-opt/genfiles/external/protobuf/src -isystem external/gif_archive/lib -isystem bazel-out/local-opt/genfiles/external/gif_archive/lib -isystem external/farmhash_archive/src -isystem bazel-out/local-opt/genfiles/external/farmhash_archive/src -isystem external/highwayhash -isystem bazel-out/local-opt/genfiles/external/highwayhash -isystem external/png_archive -isystem bazel-out/local-opt/genfiles/external/png_archive -isystem external/zlib_archive -isystem bazel-out/local-opt/genfiles/external/zlib_archive -isystem external/local_config_cuda/cuda -isystem bazel-out/local-opt/genfiles/external/local_config_cuda/cuda -isystem external/local_config_cuda/cuda/include -isystem bazel-out/local-opt/genfiles/external/local_config_cuda/cuda/include -isystem external/grpc -isystem bazel-out/local-opt/genfiles/external/grpc -isystem external/grpc/include -isystem bazel-out/local-opt/genfiles/external/grpc/include -isystem external/nanopb_git -isystem bazel-out/local-opt/genfiles/external/nanopb_git -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c native_client/c_speech_features/c_speech_features.c -o bazel-out/local-opt/bin/native_client/_objs/deepspeech/native_client/c_speech_features/c_speech_features.pic.o): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
native_client/c_speech_features/c_speech_features.c: In function 'csf_lifter':
native_client/c_speech_features/c_speech_features.c:144:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < aNFrames; i++) {
   ^
native_client/c_speech_features/c_speech_features.c:144:3: note: use option -std=c99 or -std=gnu99 to compile your code
native_client/c_speech_features/c_speech_features.c:145:5: error: 'for' loop initial declarations are only allowed in C99 mode
     for (int j = 0; j < aNCep; j++) {
     ^
native_client/c_speech_features/c_speech_features.c: In function 'csf_powspec':
native_client/c_speech_features/c_speech_features.c:267:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < aNFrames; i++) {
   ^
native_client/c_speech_features/c_speech_features.c:268:5: error: 'for' loop initial declarations are only allowed in C99 mode
     for (int j = 0; j < fft_out; j++) {
     ^
Target //native_client:deepspeech failed to build
INFO: Elapsed time: 0.532s, Critical Path: 0.31s
```

I have modified `native_client/Makefile` to include `CFLAGS ?= -O2 -Wall -std=gnu99`, but this doesn't seem to percolate up how I would expect it to.   I am using `gcc  4.8.4`.

**Comment by kdavis-mozilla** [comment]
@gvoysey I've assigned @Cwiiis who knows most about this section of the code. He can start looking at it tomorrow when he's back from vacation.

**Comment by Cwiiis** [comment]
Will look at this when I'm back (on the 20th, as opposed to tomorrow :))

**Comment by gvoysey** [comment]
@Cwiiis @kdavis-mozilla 👍 .  The only thing I'm doing today is shoveling anyway...

**Comment by gvoysey** [comment]
some further information: 
`bazel build --copt='-std=gnu99' -c opt //native_client:deepspeech`  works.  `--copt='-std=c99'` does _not_ work, and errors out fairly early. 

I believe this resolves this ticket, so i'm self-closing.

**Comment by kdavis-mozilla** [comment]
@Cwiiis Should we add this "gotcha" to the README.md?

**Comment by Cwiiis** [comment]
@kdavis-mozilla yes, I think we should. I assume either later versions of GCC don't fail on this, or we have slightly different versions of bazel, but I don't see this hurting. I'll verify on Monday and amend the readme accordingly.

**Comment by gvoysey** [comment]
forgot to mention... bazel 0.4.4.

**Comment by lissyx** [comment]
I did hit the same issue while cross-building for RPi with with 4.8 and 4.9 GCC toolchains. I guess a better fix than README would be to add `copts = ["-std=c99"]` within the BUILD file. I am in the process of making sure that this is working.

**Comment by lissyx** [comment]
Update @Cwiiis: using `copts = ["-std=gnu99"]` did work for me, with GCC 4.9.3 from rpi toolchain.

**Comment by gvoysey** [comment]
@lissyx i had made that change, but to no effect.  Passing the options directly to bazel did the trick, but threw a whole lot of "gnu99 is not a valid C++ thing" warnings early in the build process.

**Comment by Cwiiis** [comment]
So this probably comes from my mix of C++ and C in the same library... I'd tried to split things previously but never managed to get it to work. I'll see if I can tidy it up somewhat, but if not, I'll just employ the fix @lissyx suggests and update the documentation to reflect what @gvoysey is seeing.

**Comment by Cwiiis** [comment]
@lissyx @gvoysey, could you try https://github.com/Cwiiis/DeepSpeech/tree/issue441-separate-build when you have a spare moment and let me know if you have any issues?

**Comment by gvoysey** [comment]
@Cwiiis  no issues here. :+1:

**Comment by Cwiiis** [comment]
Fixed by PR #449.
